code/modules/recycling/conveyor2.dm

```diff
@@ -25,15 +25,18 @@
 	// create a conveyor
 /obj/machinery/conveyor/initialize(mapload, newdir, on = 0)
 	. = ..()
+
+
 	if(newdir)
 		set_dir(newdir)
 
-	if(dir & (dir-1)) // Diagonal. Forwards is *away* from dir, curving to the right.
-		forwards = turn(dir, 135)
-		backwards = turn(dir, 45)
-	else
-		forwards = dir
-		backwards = turn(dir, 180)
+	spawn(5)
```

Comment:
spawn is banned use addtimer if you really can't use lateinitialize and i don't know why you'd need this anyways just separate syncing directions to another proc and then override setdir

Reply:
Dude I don't even entirely know where the fuck set_dir is or if it's supposed to do literally anything but set the dir. Makes me a little cautious about doing things with it. Nor do I know all the new functions we're supposed to work with in this codebase, though the ones you mentioned sound rather self-explanatory. But I'll see about modifying it then, sometime when I can, and also bothering you about the former. It's not a hard fix, methinks.